x/staking/types/historical_info.go

```diff
@@ -8,14 +8,18 @@ import (
 
 	"github.com/cosmos/cosmos-sdk/codec"
 	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
 // NewHistoricalInfo will create a historical information struct from header and valset
 // it will first sort valset before inclusion into historical info
-func NewHistoricalInfo(header tmproto.Header, valSet Validators) HistoricalInfo {
+func NewHistoricalInfo(header tmproto.Header, valSet Validators, powerReduction sdk.Int) HistoricalInfo {
 	// Must sort in the same way that tendermint does
-	sort.Sort(ValidatorsByVotingPower(valSet))
+	// TODO: check with sort.Sort(ValidatorsByVotingPower(valSet))
```

Comment:
HistoricalInfo is only for querying and I don't think it's mandatory to change when powers are the same.
But if you want to sort as same as tendermint does, I recommend sorting by validator address as well as a secondary sort rule. 
```
		addrI, errI := valz[i].GetConsAddr()
		addrJ, errJ := valz[j].GetConsAddr()
		// If either returns error, then return false
		if errI != nil || errJ != nil {
			return false
		}
		return bytes.Compare(addrI, addrJ) == -1
```